src/test/utils.ts

```diff
@@ -109,3 +110,150 @@ export function toRange(
   const end = new vscode.Position(eLine, eChar)
   return new vscode.Range(start, end)
 }
+
+// Add this function to normalize line endings
+function eolReplacer(key: string, value: any) {
+  if (typeof value === 'string') {
+    return value.replace(/\r\n/g, '\n')
+  }
+  return value
+}
+
+export function assertCodeActionArraysEqual(
+  codeActions1: vscode.CodeAction[],
+  codeActions2: vscode.CodeAction[],
+  docUri: vscode.Uri
+): void {
+  assert.equal(codeActions1.length, codeActions2.length)
+  codeActions1.forEach((codeAction1, i) => {
+    const codeAction2 = codeActions2[i]
+    assertCodeActionsEqual(codeAction1, codeAction2, docUri)
+  })
+}
+export function assertCodeActionsEqual(
+  codeAction1: vscode.CodeAction,
+  codeAction2: vscode.CodeAction,
+  docUri: vscode.Uri
+): void {
+  if (codeAction1.title !== codeAction2.title) {
+    throw new Error(
+      `Code Action Title: ${codeAction1.title} !== ${codeAction2.title}`
+    )
+  }
+
+  if (codeAction1.isPreferred !== codeAction2.isPreferred) {
+    throw new Error(
+      `Code Action isPreferred: ${codeAction1.isPreferred} !== ${codeAction2.isPreferred}`
+    )
+  }
+
+  if (codeAction1.edit === undefined && codeAction2.edit === undefined) {
+    return // both are undefined which counts as equal
+  }
+  if (codeAction1.edit === undefined) {
+    throw new Error(
+      `The First Code Action's edit is undefined, but the second is ${codeAction2.edit}`
+    )
+  }
+  if (codeAction2.edit === undefined) {
+    throw new Error(
+      `The First Code Action's edit is ${codeAction1.edit}, but the second is undefined`
+    )
+  }
+  // both exist
+  let edits1 = codeAction1.edit!.get(docUri)
+  let edits2 = codeAction2.edit!.get(docUri)
+  edits1.forEach((_, i) => {
+    let text1 = JSON.stringify(edits1[i], eolReplacer, 2)
```

Comment:
do we still have eol issue after setting vscode preference?  If there is no issue, let's keep the test code simple

Reply:
You're right.

I've removed this for now as it should be addressed in a separate PR if at all.
In this PR, the eol issue is properly handled by the `activate` function in `src/test/utils.ts`
```ts
doc = await vscode.workspace.openTextDocument(getDocUri('test.ts'))
editor = await vscode.window.showTextDocument(doc)
// enforce unix line endings, otherwise tests fail on windows
await editor.edit(builder => {
  builder.setEndOfLine(vscode.EndOfLine.LF)
})
```

FYI: In my other PR I found that adding the `eol` property to `.vscode/settings.json` did **not** resolve the issue.